--- mesa/mesa-20.2.0/meson.build
+++ mesa-20.2.0/meson.build
@@ -1597,7 +1597,7 @@
     prog_bison = find_program('yacc', 'bison', required : with_any_opengl)
   endif
 else
-  prog_bison = find_program('bison', required : with_any_opengl)
+  prog_bison = find_program('bison', 'byacc', required : with_any_opengl)
 
   # Disable deprecated keyword warnings, since we have to use them for
   # old-bison compat.  See discussion in
--- mesa/mesa-20.2.0/src/compiler/glsl/glcpp/glcpp-lex.l
+++ mesa-20.2.0/src/compiler/glsl/glcpp/glcpp-lex.l
@@ -27,7 +27,7 @@
 #include <ctype.h>
 
 #include "glcpp.h"
-#include "glcpp-parse.h"
+#include "glcpp-parse.tab.h"
 
 /* Flex annoyingly generates some functions without making them
  * static. Let's declare them here. */
--- mesa/mesa-20.2.0/src/compiler/glsl/glcpp/meson.build
+++ mesa-20.2.0/src/compiler/glsl/glcpp/meson.build
@@ -19,12 +19,11 @@
 # SOFTWARE.
 
 glcpp_parse = custom_target(
-  'glcpp-parse.[ch]',
+  'glcpp-parse.tab.[ch]',
   input : 'glcpp-parse.y',
-  output : ['glcpp-parse.c', 'glcpp-parse.h'],
+  output : ['glcpp-parse.tab.c', 'glcpp-parse.tab.h'],
   command : [
-    prog_bison, '-o', '@OUTPUT0@', '-p', 'glcpp_parser_',
-    '--defines=@OUTPUT1@', '@INPUT@',
+    prog_bison, '-p', 'glcpp_parser_', '-d', '-b', '@OUTDIR@/@BASENAME@', '@INPUT@',
   ],
 )
 
--- mesa/mesa-20.2.0/src/compiler/glsl/glsl_lexer.ll
+++ mesa-20.2.0/src/compiler/glsl/glsl_lexer.ll
@@ -26,7 +26,7 @@
 #include "util/strtod.h"
 #include "ast.h"
 #include "glsl_parser_extras.h"
-#include "glsl_parser.h"
+#include "glsl_parser.tab.h"
 #include "main/mtypes.h"
 
 static int classify_identifier(struct _mesa_glsl_parse_state *, const char *,
--- mesa/mesa-20.2.0/src/compiler/glsl/glsl_parser.yy
+++ mesa-20.2.0/src/compiler/glsl/glsl_parser.yy
@@ -98,7 +98,7 @@
 %lex-param   {struct _mesa_glsl_parse_state *state}
 %parse-param {struct _mesa_glsl_parse_state *state}
 
-%union {
+%union YYSTYPE {
    int n;
    int64_t n64;
    float real;
--- mesa/mesa-20.2.0/src/compiler/glsl/glsl_parser_extras.cpp
+++ mesa-20.2.0/src/compiler/glsl/glsl_parser_extras.cpp
@@ -36,7 +36,7 @@
 #include "util/mesa-sha1.h"
 #include "ast.h"
 #include "glsl_parser_extras.h"
-#include "glsl_parser.h"
+#include "glsl_parser.tab.h"
 #include "ir_optimization.h"
 #include "loop_analysis.h"
 #include "builtin_functions.h"
--- mesa/mesa-20.2.0/src/compiler/glsl/meson.build
+++ mesa-20.2.0/src/compiler/glsl/meson.build
@@ -23,10 +23,10 @@
 glsl_parser = custom_target(
   'glsl_parser',
   input : 'glsl_parser.yy',
-  output : ['glsl_parser.cpp', 'glsl_parser.h'],
+  output : ['glsl_parser.tab.cpp', 'glsl_parser.tab.h'],
   command : [
-    prog_bison, '-o', '@OUTPUT0@', '-p', '_mesa_glsl_', '--defines=@OUTPUT1@',
-    '@INPUT@',
+    prog_bison, '-p', '_mesa_glsl_', '-b', '@OUTDIR@/@BASENAME@', '-d',
+    '-o', '@OUTPUT0@', '@INPUT@',
   ],
 )
 
--- mesa/mesa-20.2.0/src/intel/tools/meson.build
+++ mesa-20.2.0/src/intel/tools/meson.build
@@ -166,8 +166,7 @@
   input : 'i965_gram.y',
   output : ['i965_gram.tab.c', 'i965_gram.tab.h'],
   command : [
-    prog_bison, '@INPUT@', '--defines=@OUTPUT1@',
-    '--output=@OUTPUT0@'
+    prog_bison, '-b', '@OUTDIR@/i965_gram', '-d', '@INPUT@'
   ]
 )
 
--- mesa/mesa-20.2.0/src/mesa/program/meson.build
+++ mesa-20.2.0/src/mesa/program/meson.build
@@ -29,5 +29,5 @@
   'program_parse_tab.[ch]',
   input : 'program_parse.y',
   output : ['program_parse.tab.c', 'program_parse.tab.h'],
-  command : [prog_bison, '-o', '@OUTPUT0@', '--defines=@OUTPUT1@', '@INPUT@'],
+  command : [prog_bison, '-b', '@OUTDIR@/@BASENAME@', '-d', '@INPUT@'],
 )
