--- render.c
+++ render.c
@@ -478,13 +478,11 @@
 #ifndef GLES2
     if (effects & EFFECT_WIREFRAME) {
         gl_update_uniform(shader, "u_l_amb", 0.0);
-        GL(glPolygonMode(GL_FRONT_AND_BACK, GL_LINE));
         if (item->size == 4)
             GL(glDrawElements(GL_TRIANGLES, item->nb_elements * 6,
                               GL_UNSIGNED_SHORT, 0));
         else
             GL(glDrawArrays(GL_TRIANGLES, 0, item->nb_elements * item->size));
-        GL(glPolygonMode(GL_FRONT_AND_BACK, GL_FILL));
         gl_update_uniform(shader, "u_l_amb", rend->settings.ambient);
     }
 #endif
@@ -953,7 +951,6 @@
         GL(glGenFramebuffers(1, &g_shadow_map_fbo));
         GL(glBindFramebuffer(GL_FRAMEBUFFER, g_shadow_map_fbo));
         #ifndef GLES2
-        GL(glDrawBuffer(GL_NONE));
         GL(glReadBuffer(GL_NONE));
         #endif
         g_shadow_map = texture_new_surface(2048, 2048, 0);
