--- configure.ac
+++ configure.ac
@@ -15,20 +15,9 @@
 AC_PROG_LIBTOOL
 AC_LANG([C])
 
-AC_PROG_INTLTOOL([0.50])
 
 AC_CANONICAL_HOST
 
-
-# Checks for libraries.
-AM_GNU_GETTEXT_VERSION([0.17])
-AM_GNU_GETTEXT([external])
-
-GETTEXT_PACKAGE=gammastep
-AC_SUBST(GETTEXT_PACKAGE)
-AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Package name for gettext])
-
-
 PKG_CHECK_MODULES([DRM], [libdrm], [have_drm=yes], [have_drm=no])
 
 PKG_CHECK_MODULES([X11], [x11], [have_x11=yes], [have_x11=no])
@@ -238,7 +227,6 @@
 
 AC_CONFIG_FILES([
 	Makefile
-	po/Makefile.in
 	src/Makefile
 	src/gammastep_indicator/Makefile
 ])
